modules/backend/formwidgets/CodeEditor.php

```diff
@@ -167,6 +167,7 @@ public function prepareVars()
         $this->vars['highlightActiveLine'] = $this->highlightActiveLine;
         $this->vars['useSoftTabs'] = $this->useSoftTabs;
         $this->vars['showGutter'] = $this->showGutter;
+        $this->vars['safeMode'] = $this->parentForm->model->isSafeMode();
```

Comment:
This can't exist here either because the code editor can be used on other models than CMS templates. You could probably implement this as a `hint` property (so `hint: cms::lang.cms_object.safe_mode_enabled`) on the codeeditor formwidget instead though and I'd be fine with that.

Reply:
done